dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/gist/GistValidator.java

```diff
@@ -127,12 +132,74 @@ private void validateField( Field f, RelativePropertyContext context )
         }
     }
 
-    private void validateFilter( Property filter )
+    private void validateFilter( Filter f, RelativePropertyContext context )
     {
+        Property filter = context.resolveMandatory( f.getPropertyPath() );
         if ( !filter.isPersisted() )
         {
             throw createIllegalProperty( filter, "Property `%s` cannot be used as filter property." );
         }
+
+        validateFilterArgument( f );
+        validateFilterAccess( f );
+    }
+
+    private void validateFilterAccess( Filter f )
+    {
+        Comparison operator = f.getOperator();
+        if ( operator.isAccessCompare() )
+        {
+            String[] ids = f.getValue();
+            if ( operator != Comparison.CAN_ACCESS && ids.length != 1 )
+            {
+                throw createIllegalFilter( f, "Filter `%s` requires a single user ID as argument." );
+            }
+            if ( operator == Comparison.CAN_ACCESS && ids.length != 2 )
+            {
+                throw createIllegalFilter( f, "Filter `%s` requires a user ID and a access pattern argument." );
+            }
+            if ( operator == Comparison.CAN_ACCESS )
+            {
+                // OBS! we include this user input directly in the query so we
+                // have
+                // to make sure it is not malicious
+                String pattern = f.getValue()[1];
+                if ( !pattern.matches( "[_%rw]{2,8}" ) )
+                {
+                    throw createIllegalFilter( f,
+                        "Filter `%s` pattern argument must be 2 to 8 letters allowing letters 'r', 'w', '_' and '%%'." );
+                }
+            }
+            User user = userService.getUser( ids[0] );
+            if ( user == null )
+            {
+                throw createIllegalFilter( f, "User for filter `%s` does not exist." );
+            }
+            if ( !aclService.canRead( currentUserService.getCurrentUser(), user ) )
```

Comment:
I don't think `User` class  is `sharable`
Maybe this could be achieved by `currentUser.canManage( user ) || currentUser == user.getCreatedBy`

Reply:
Done.